ewc-affiliate/install-validator-centos-7-production-nethermind.sh

```diff
@@ -0,0 +1,493 @@
+wget $CHAINSPEC_URL -O chainspec/energyweb.json
+
+echo "Creating Account..."
+
+# Generate random account password and store
+XPATH="$(pwd)"
+PASSWORD="$(openssl rand -hex 32)"
+echo "$PASSWORD" > .secret
+chmod 400 .secret
+chown 1000:1000 .secret
+mv .secret keystore/.secret
+
+docker run -d --network host --name nethermind \
+    -v ${XPATH}/keystore/:/nethermind/keystore \
+    ${NETHERMIND_VERSION} --config ${CHAINNAME} --Init.EnableUnsecuredDevWallet true --JsonRpc.Enabled true
+    
+generate_account_data() 
+{
+cat << EOF
+{ "method": "personal_newAccount", "params": ["$PASSWORD"], "id": 1, "jsonrpc": "2.0" }
+EOF
+}
+
+echo "Waiting 30 sec for nethermind to come up and create an account..."
+sleep 30
+# Send request to create account from seed
+ADDR=`curl --request POST --url http://localhost:8545/ --header 'content-type: application/json' --data "$(generate_account_data)" | jq -r '.result'`
+
+echo "Account created: $ADDR"
+INFLUX_USER=${ADDR:2} # cutting 0x prefix
+INFLUX_PASS="$(openssl rand -hex 16)"
+
+
+# got the key now discard of the nethermind instance
+docker stop nethermind
+docker rm -f nethermind
+
+writeNethermindConfig
+NETHERMIND_KEY_FILE="$(ls -1 ./keystore/|grep UTC|tail -n1)"
+
+# Prepare nethermind telemetry pipe
+mkfifo /var/spool/nethermind.sock
+chown telegraf /var/spool/nethermind.sock
+
+# Write NLog config file
+wget $NLOG_CONFIG -O NLog.config
+setJsonRpcLogsLevelToError
+
+# Write the docker-compose file to disk
+writeDockerCompose
+
+# start everything up
+docker-compose up -d
+
+# Collect the enode from nethermind over RPC
+
+echo "Waiting 15 sec for nethermind to come up and generate the enode..."
+sleep 15
+ENODE=`curl -s --request POST --url http://localhost:8545/ --header 'content-type: application/json' --data '{ "method": "net_localEnode", "params": [], "id": 1, "jsonrpc": "2.0" }' | jq -r '.result'`
+
+# Now all information is complete to write the telegraf file
+writeTelegrafConfig
+service telegraf restart
+
+echo "Setting up firewall"
+
+systemctl enable iptables
+
+# non-docker services
+iptables -F INPUT
+iptables -F DOCKER-USER || true
+iptables -X FILTERS || true
+iptables -N FILTERS || true
+iptables -A FILTERS -p tcp --dport 2222 -j ACCEPT  # SSH
+iptables -A FILTERS -p tcp --dport 30303 -j ACCEPT # P2P tcp
+iptables -A FILTERS -p udp --dport 30303 -j ACCEPT # P2P udp
+iptables -A FILTERS -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
+iptables -A FILTERS -j DROP
+
+# hook filters chain to input and docker
+iptables -A INPUT -p icmp --icmp-type any -j ACCEPT
+iptables -A INPUT -i lo -j ACCEPT
+iptables -A INPUT -j FILTERS
+iptables -P INPUT DROP
+
+iptables -A DOCKER-USER -i $NETIF -j FILTERS
+iptables -A DOCKER-USER -j RETURN
+service iptables save
+
+# run automated post-install audit
+cd /opt/
+wget https://downloads.cisofy.com/lynis/lynis-2.7.1.tar.gz
+tar xvzf lynis-2.7.1.tar.gz
+mv lynis /usr/local/
+ln -s /usr/local/lynis/lynis /usr/local/bin/lynis
```

Comment:
Since `/usr/local/bin` isn't in env PATH by default we can either run `export PATH=$PATH:/usr/local/bin` or `ln -s /usr/local/lynis/lynis /usr/bin/lynis`

Reply:
agreed, fixed.